enhancements/update/versioning-cincinnati-api.md

```diff
@@ -0,0 +1,168 @@
+versioning Cincinnati payloads by media type, for example `application/vnd.redhat.cincinnati.v1+json`.
+The move to supplying requested version in media type field will require us to move towards a generic un-versioned Graph
+URI, with the move, we'll still support old URI for foreseeable future.
+
+## Motivation
+
+### Goals
+1. Embed version information in json schema.
+2. Implement media type versioning for clients requesting a specific schema version of Cincinnati
+3. Move to a generic un-versioned URI for Cincinnati
+4. Cincinnati continues to support older clients after adding support for new clients for foreseeable future.
+5. Add metrics to identify hits on URIs and versions.
+
+### Non-Goals
+1. This enhancement does not attempt to drop the `v1` URI.
+
+
+## Proposal
+
+### User Stories
+
+### Implementation
+
+#### Version information in json schema
+Embed the version information in json schema. The updated json schema should look like:
+```json
+{
+    "version": 1,
+    "nodes": [],
+    "edges": []
+}
+```
+
+#### Media Type Versioning
+Currently, the versioning is done by using the versioned URI and does not support media type versioning.
+This enhancement allows users to request a specific version of the Cincinnati specification
+[proactively][proactive-negotiation] with [the `Accept` header][accept].  
+For example, `Accept: application/vnd.redhat.cincinnati.v1+json` says "I only understand v1 Cincinnati graphs;
+please give me that format if possible".  
+In the event where the media type is not provided, or `application/json` is provided, Cincinnati should return a default
+response that is compatible with the oldest supported version so as not to break old clusters from upgrading.
+
+In a similar space, the Open Container Initiative Image Format Specification
+[uses versioned media types][image-spec-media-types] to identify versions of configuration and other formats.
+
+
+#### URI Change
+The current URI follows `{URL}/v1/graph`. The new un-versioned URI will follow `{URL}/graph`. The existing URI can be
+redirected to the new URI. This can be done through cincinnati. In the case where an old URI is being called, cincinnati
+will by default return the graph that is compatible with the oldest supported version.
+
+
+#### Version metrics
+Cincinnati should also implement a way to understand how many clients are consuming the various versions.
+This can be done by implementing metrics for old URI vs new URI in Cincinnati. Similarly, metrics should be implemented
+to understand the consumption of various media types (versions). This will help us to make decisions on deprecating the
+old versions and URI. For example, a single `cincinnati_pe_http_response_total` counter with labels for `uri` and `version`
+
+
+### Risks and Mitigations
+* As there are some customers that have old URI hardcoded while creating the cluster, we will have to support old URI
```

Comment:
Should we consider installer validation to block new installs using the versioned URI when the client expects a newer version than v1? For example if 4.10 CVO uses a v2 API we don't want to allow them to inadvertantly downrev to v1 and lose targetted edge blocking due to their configuration.

Reply:
we are moving away from versioned URI to unversioned. So all requests from `/v1/graph` will be handled as if they're coming to `/graph`. Going forward we will be accepting the version using Media Type Versioning. 
So, if someone hits the old URI with a correct media type version, they'll end up getting the intended graph or if someone hits the new URI without the media type version, they'll end up getting the oldest supported graph.  
Installer validation will speed up the process of shifting to the new URI!

Reply:
So the conclusion is that we'd never get rid of the legacy URL. That's fine.